evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -620,3 +621,32 @@ class CustomPipeline(MulticlassClassificationPipeline):
 
     automl = AutoMLSearch(problem_type="multiclass", allowed_pipelines=[CustomPipeline], n_jobs=1)
     automl.search(X, y)
+
+
+def test_automl_binary_nonlinear_pipeline_search(nonlinear_binary_pipeline_class, X_y_binary):
```

Comment:
Since we're not mocking score or fit, can you make sure `n_jobs=1` for these estimators? I think you can add a method/modify `safe_init_pipeline_with_njobs_1` in the Helpers test fixture!